### PR TITLE
TST/VIZ: add test for legend colors for DataFrame with duplicate column labels #11136

### LINKED ISSUE #11136: Plot uses incorrect colors when columns have the same name
It seems like the `plot` function jumbles up the colors when columns in a DataFrame have the same name:

``` python
pd.concat([pd.DataFrame({'b': [0, 1, 0], 'a': [1, 2, 3]}), pd.DataFrame({'a': [2, 4, 6]})], axis=1).plot();
```

image

### PR DESCRIPTION
**Pull request by gabrielvf1**
- [X] closes #11136
- [X] tests added / passed
- [X] passes `black pandas`
- [X] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/plotting/test_frame.py [hunk 1]
```diff
@@ -3316,6 +3316,22 @@ def test_missing_markers_legend_using_style(self):
         self._check_legend_labels(ax, labels=["A", "B", "C"])
         self._check_legend_marker(ax, expected_markers=[".", ".", "."])
 
+    def test_colors_of_columns_with_same_name(
```
Comment (simonjayhawkins) [inline pandas/tests/plotting/test_frame.py]: something like test_legend_colors_duplicate_labels ?

**Review by simonjayhawkins — COMMENTED** [review]
Thanks @gabrielvf1 for the PR.

File: pandas/tests/plotting/test_frame.py (hunk 1 above)
Comment (gabrielvf1) [reply pandas/tests/plotting/test_frame.py]: Thanks for the reply, i have just made the changes . Can you check if is everything right now?

**Comment by gabrielvf1** [comment]
I have just made the changes. Can you check if is everything right now? Really thanks!

**Review by simonjayhawkins — COMMENTED** [review]
Thanks @gabrielvf1 for those updates.

**Comment by gabrielvf1** [comment]
Fix were made, the for syntax is better now and now im using methods from matplotlib like get_color() and get_legend().

**Review by simonjayhawkins — APPROVED** [review]
Thanks @gabrielvf1 lgtm pending green

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
Great thanks @gabrielvf1

**Pull request merged**